march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py

```diff
@@ -19,13 +20,15 @@ def __init__(self, view, robot):
         self.playback_speed = 100
         self.time_slider_thread = None
         self.current_time = 0
-
         self.robot = robot
-        self.subgait = ModifiableSubgait.empty_subgait(self, self.robot)
+
+        empty_subgait_file = os.path.join(rospkg.RosPack().get_path('march_rqt_gait_generator'),
+                                          'resource/empty_gait/empty_subgait/empty_subgait.subgait')
+        self.subgait = ModifiableSubgait.from_file(robot, empty_subgait_file, self)
         self.settings_changed_history = RingBuffer(capacity=100, dtype=list)
         self.settings_changed_redo_list = RingBuffer(capacity=100, dtype=list)
 
-        standing = ModifiableSubgait.empty_subgait(self, self.robot)
+        standing = ModifiableSubgait.from_file(robot, empty_subgait_file, self)
```

Comment:
Why read twice from file, if you can just copy the empty subgait?

Reply:
That is unfortunately not so simple, because the ModifiableSubgait contains a reference to the GaitGeneratorController, which cannot be copied. When you mentioned it, I realised that the subgait "standing" should not be a ModifiableSubgait at all, I changed it to a Subgait. I still read the file twice, but I fear that that is unavoidable without some considerable refactoring.